library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,425 @@
+      end
+
+      res = {
+        "widget"        => :combobox,
+      }
+      res["items"] = items if respond_to?(:items)
+
+      res.merge(super)
+    end
+  end
+
+  class SelectionBoxWidget < AbstractionWidget
+    # description for selectionbox additionally support `items` method.
+    # `items` method have to return array of string pairs, where first value is
+    # item id and second is item label.
+    # @example items method
+    #   def items
+    #     [
+    #       [ "Canada", _("Canada")],
+    #       [ "USA", _("United States of America")],
+    #       [ "North Pole", _("Really cold place")],
+    #     ]
+    #   end
+    #
+    def description
+      unless respond_to?(:label)
+        raise "For input field widget '#{self.class}' label method have to be defined"
+      end
+
+      res = {
+        "widget"        => :selection_box, # yeah, really so incosistent to have combox and selection_box
+      }
+      res["items"] = items if respond_to?(:items)
+
+      res.merge(super)
+    end
+
+  protected
+    def value
+      Yast::UI.QueryWidget(Id(widget_id), :CurrentItem)
+    end
+
+    def value=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :CurrentItem, val)
+    end
+  end
+
+  class MultiSelectionBoxWidget < AbstractionWidget
+    # description for multiselectionbox additionally support `items` method.
+    # `items` method have to return array of string pairs, where first value is
+    # item id and second is item label.
+    # @example items method
+    #   def items
+    #     [
+    #       [ "Canada", _("Canada")],
+    #       [ "USA", _("United States of America")],
+    #       [ "North Pole", _("Really cold place")],
+    #     ]
+    #   end
+    #
+    def description
+      unless respond_to?(:label)
+        raise "For input field widget '#{self.class}' label method have to be defined"
+      end
+
+      res = {
+        "widget"        => :multi_selection_box,
+      }
+      res["items"] = items if respond_to?(:items)
+
+      res.merge(super)
+    end
+
+  protected
+    def values
+      Yast::UI.QueryWidget(Id(widget_id), :SelectedItems)
+    end
+
+    def values=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :SelectedItems, val)
+    end
+  end
+
+  class IntField < AbstractionWidget
+    include ValueBasedWidget
+
+    # description for combobox additionally support `minimum` and `maximum` methods.
+    # Both methods have to FixNum, where it is limited by C signed int range (-2**30 to 2**31-1).
+    # @example minimum and maximum methods
+    #   def minimum
+    #     50
+    #   end
+    #
+    #   def maximum
+    #     200
+    #   end
+    #
+    def description
+      unless respond_to?(:label)
+        raise "For input field widget '#{self.class}' label method have to be defined"
+      end
+
+      res = {
+        "widget"        => :intfield,
+      }
+      res["minimum"] = minimum if respond_to?(:minimum)
+      res["maximum"] = minimum if respond_to?(:maximum)
```

Comment:
It is question if we want it as own API or just layer over CWM, which adds value, but do not modify defaults